torch/csrc/jit/codegen/cuda/test/test_gpu_rng.cu

```diff
@@ -106,32 +106,33 @@ at::Tensor generate_uniform(int64_t size, at::ScalarType dtype) {
 } // namespace
 
 TEST_F(NVFuserTest, FusionRNGValidateWithCURand_CUDA) {
-  for (int64_t size : {16, 1024, 10001, 10002, 10003, 100000, 10000001}) {
-    for (auto dtype : {kFloat, kDouble}) {
-      std::unique_ptr<Fusion> fusion_ptr = std::make_unique<Fusion>();
-      auto fusion = fusion_ptr.get();
-      FusionGuard fg(fusion);
+  std::unique_ptr<Fusion> fusion_ptr = std::make_unique<Fusion>();
+  auto fusion = fusion_ptr.get();
+  FusionGuard fg(fusion);
 
-      Int* size_val = IrBuilder::create<Int>();
-      fusion->addInput(size_val);
-      TensorView* tv0 = rand({size_val}, aten_to_data_type(dtype));
-      fusion->addOutput(tv0);
+  Int* size_val = IrBuilder::create<Int>();
+  fusion->addInput(size_val);
+  TensorView* tv0 = rand({size_val}, DataType::Float);
+  TensorView* tv1 = rand({size_val}, DataType::Double);
```

Comment:
This will be segmented because we do not support pointwise horizontal fusion today.

Reply:
Yeah, not segmenting something like this could take some work, we'd have to pipe through some additional id mapping constraints or do some analysis to understand this could be fused.